crates/sui-core/src/authority.rs

```diff
@@ -941,15 +941,8 @@ impl AuthorityState {
             .start_timer();
         self.metrics.tx_orders.inc();
 
-        // The should_accept_user_certs check here is best effort, because
-        // between a validator signs a tx and a cert is formed, the validator
-        // could close the window.
-        if !epoch_store
-            .get_reconfig_state_read_lock_guard()
-            .should_accept_user_certs()
-        {
-            return Err(SuiError::ValidatorHaltedAtEpochEnd);
-        }
+        // Ensure that validator cannot reconfigure while we are signing the tx
+        let _execution_lock = self.execution_lock_for_signing().await;
```

Comment:
can this lock be moved into `handle_transaction_impl` so it also works with #19401